biolink/pieces-parts/rank-regulators-exe.rkt

```diff
@@ -0,0 +1,127 @@
+         be inaccurate due to inaccuracies in NLP generated knowledge graphs. After using both querying
+         methods, we compare the two results to find intersections between them. The results that are 
+         common between both methods are held in higher confidence, but that is not to say that results
+         that don't lie in the overlap are inaccurate.
+
+Input/Output: Takes a list of the curies of genes of interest and outputs a sorted association list of
+              gene to regulated gene relationships. The association list is sorted from most to least
+              by the number of other genes/gene products that a gene regulates.
+
+Information in Output: Contained within the sorted association list that is returned is a pair made from
+                       cons-ing the regulator gene's (G) curie and name. These (G curie. G name) pairs are
+                       the keys of the assoc list. The value is a list of lists that contains the regulated
+                       genes (rG) that the regulator gene (G) regulates along with additional information.
+                       Each element in the value is a list that contains the rG's curie, name, the presumed
+                       direction of regulation (told by predicates), and provenance for the edges in the form
+                       of pubmed ids. The functions 1-hop-query/report and find-intersections also include
+                       a boolean that indicates whether or not the G and rG directly interact with each other
+                       in the body (#t if yes) and provenance for that information also in the form of pubmed
+                       ids.
+
+Note: The functions called in this file are more extensively documented in rank-regulators.rkt
+|#
+
+#|
+The following calls are an example based off an ARDS gene list, defined as ards-genes. There are 517 ENSEMBL
+gene curies on that list, and they yield 468 uniprot curies.
+
+Note: runtimes given in comments below are based off a 2020 MacBook Pro (2 GHz Quad-Core Intel Core i5,
+      16G RAM). They are meant to give an idea of which calls take a lot longer to finish.
+|#
+
+#|
+Turns the inputted gene list of interest into a list of uniprot curies in order to be passed
+into the GO query.
+runtime: < 30s
+|#
+;;function call: genes->uniprots | takes: list of gene curies | returns: list of uniprot curies
+(define uniprots (time (genes->uniprots ards-genes)))
+
+#|
+The next two statements 1) run the GO query and 2) reports the results of the query in an association list.
+runtime:
+    GO-query-edges: ~5 min
+    go-query-results: just under 2 hrs
+|#
+;;function call: query-GO | takes: list of uniprot curies | returns: list of three edge hashes
+(define GO-query-edges (time (query-GO uniprots)))
+
+;;function call: report-GO-queries | takes: list of three edge hashes | returns: sorted assoc list
+(define go-query-results (time (report-GO-queries GO-query-edges)))
```

Comment:
I get an error here:
`; assoc: not a proper list: #<void> [,bt for context]`

@mzheng17

Reply:
I think this is a quick fix:
```
diff --git a/biolink/pieces-parts/rank-regulators.rkt b/biolink/pieces-parts/rank-regulators.rkt
index aa98e0c..e628504 100644
--- a/biolink/pieces-parts/rank-regulators.rkt
+++ b/biolink/pieces-parts/rank-regulators.rkt
@@ -316,7 +316,7 @@ function) because that edge is the one that has the directional predicates.
                                (lambda (v)
                                  (cond
                                    [(assoc (car new-val) v) => (lambda (a) 
-                                                                 (unless (empty? pubmeds)
+                                                                 (if (empty? pubmeds) '()
                                                                    (set-add (set-remove v a)
                                                                             (list (car a)
                                                                                   (set-union (cadr a) pubmeds)))))]
```

@mzheng17

Reply:
Maybe `v` instead of `'()` would be more appropriate. I am lacking a bit of context here.

@mzheng17